src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -700,14 +700,29 @@ gl_sarray drawing_classifier::predict(gl_sframe data, std::string output_type) {
     result = result.apply(max_prob_label, class_labels.front().get_type());
 
   } else if (output_type == "probability") {
-    auto max_prob = [=](const flexible_type& ft) {
+    /** The output_type="probability" is to provide the probability of the True
+     *  class in binary classifiers.
+     *  For example, if one is building a binary classifier for
+     *  "cat" vs "not cat", output_type="probability" should output the
+     *  probability of the data point being "cat", not the probability of the
+     *  predicted class.
+     */
+
+    size_t num_classes = read_state<flex_int>("num_classes");
+    DASSERT_GT(num_classes, 0);
+
+    if (num_classes > 2) {
+      log_and_throw(
+          "Use probability_vector in case of multi-class classification.");
+    }
+
+    auto true_class_probability = [=](const flexible_type& ft) {
       const flex_vec& prob_vec = ft.get<flex_vec>();
-      auto max_it = std::max_element(prob_vec.begin(), prob_vec.end());
-      return *max_it;
+      size_t last_index = num_classes - 1;  // either 0 or 1
```

Comment:
Could you explain why only the last probability is returned?

Reply:
With True and False labels, i.e. "cat" and "not cat", we assume that the False label has index 0 and the True label has index 1. I'm not sure where we broadcast this to users (probably somewhere).

Reply:
The relative order of `cat` and `not-cat` should depend on the label from the training data. set, i.e, data iterator may return label vector of <cat, not-cat> or <cat, not-cat>. You may. check where `num_classes` is introduced in `init_training`.